packages/db/src/repositories/perps.ts

```diff
@@ -443,6 +444,118 @@ export class PerpsRepository {
     }
   }
 
+  /**
+   * Get agents who fail daily volume requirements using SQL aggregation
+   * Calculates volume delta and requirements entirely in database
+   *
+   * Base equity calculation uses hybrid approach:
+   * - Max(initialCapital, periodStartEquity * 0.8)
+   * - The 0.8 multiplier (80% floor) protects against loss exploitation
+   * - Agents who lose money can't reduce requirements below initial capital
+   * - Agents who gain money have requirements scale with 80% of period start equity
+   *
+   * @param competitionId Competition ID
+   * @param checkTimestamp Current timestamp for the check
+   * @param volumeMultiplier Daily volume multiplier (e.g., 0.75)
+   * @returns Array of agent IDs who should be disqualified
+   */
+  async getAgentsWithInsufficientDailyVolume(
+    competitionId: string,
+    checkTimestamp: Date,
+    volumeMultiplier: number,
+  ): Promise<string[]> {
+    try {
+      const twentyFourHoursAgo = new Date(
+        checkTimestamp.getTime() - 24 * 60 * 60 * 1000,
+      );
+
+      // Hybrid base equity calculation constants
+      const PERIOD_START_EQUITY_MULTIPLIER = 0.8; // 80% floor protects against loss exploitation
+
+      // Get active agents subquery
+      const activeAgents = this.getActiveAgentsSubquery(competitionId);
+
+      // Latest summary subquery (current)
+      const latestSummarySubquery = this.#dbRead
+        .select({
+          agentId: perpsAccountSummaries.agentId,
+          currentVolume: perpsAccountSummaries.totalVolume,
+          currentEquity: perpsAccountSummaries.totalEquity,
+          initialCapital: perpsAccountSummaries.initialCapital,
+        })
+        .from(perpsAccountSummaries)
+        .where(
+          and(
+            eq(perpsAccountSummaries.competitionId, competitionId),
+            sql`${perpsAccountSummaries.agentId} = ${activeAgents.agentId}`,
```

Comment:
Why use the sql`` string template here instead of `eq(perpsAccountSummaries.agentId, activeAgents.agentId`

Reply:
For lateral join correlations in Drizzle ORM, I believe we have to use `sql` templates rather than `eq()` since`sql` templates for cross-scope column references, but please let me know if there's a pattern I'm missing